lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -424,7 +424,7 @@ def ipv4_address(vm)
     # Multiple reboots occur during guest customization in which a link-local
     # address is assigned. As such, we need to wait until a routable IP address
     # becomes available. This is most commonly an issue with Windows instances.
-    sleep 2 while vm.guest.net[bootstrap_nic_index].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.origin == 'linklayer'
+    sleep 2 while (vm.guest.net[bootstrap_nic_index].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.nil? || vm.guest.net[bootstrap_nic_index].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.origin == 'linklayer')
```

Comment:
Not really sure how to write that differently, it needs to do the full invocation so it hits the API during each repetition of the sleep, so we can't make it a variable earlier :(